### PR TITLE
Share sytest clone across runs

### PR DESCRIPTION
**Pull request by illicitonion**
Depends on https://github.com/matrix-org/synapse/pull/380

### CODE REVIEW

File: jenkins.sh [hunk 1]
```diff
@@ -1,4 +1,49 @@
...
+else
+  cd .sytest-base
+  git fetch
+  cd ..
+fi
+
+rm -rf sytest
+git clone .sytest-base sytest --shared
+cd sytest
+git remote add upstream https://github.com/matrix-org/sytest.git
```
Comment (NegativeMjark) [inline jenkins.sh]: Won't this fail if a "remote" called upstream already exists?

Comment (NegativeMjark) [inline jenkins.sh]: Ah it's in its own clone of upstream.

File: jenkins.sh [hunk 2]
```diff
@@ -1,4 +1,49 @@
...
+tox
+
+: ${GIT_BRANCH:="origin/$(git rev-parse --abbrev-ref HEAD)"}
+
+set +u
+. .tox/py27/bin/activate
+set -u
+
+if [[ ! -e .sytest-base ]]; then
+  git clone https://github.com/matrix-org/sytest.git .sytest-base
```
Comment (NegativeMjark) [inline jenkins.sh]: I wonder if the --mirror=fetch makes sense here.
Then we might be able to avoid need to faff around adding a remote below.

Comment (illicitonion) [inline jenkins.sh]: I leave this as an exercise to the reviewer :)

**Comment by NegativeMjark** [comment]
I've fixed up the git clone to be a bit simpler. Hopefully this still does what @illicitonion wanted.

**Pull request merged**